rids/observations.py

```diff
@@ -12,31 +12,28 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""
-Tests for the network_capture library.
-"""
 
+from dataclasses import dataclass
+import ipaddress
+from typing import Union
 
-import io
-import pytest
 
-from rids import network_capture
+IpType = Union[ipaddress.IPv4Address, ipaddress.IPv6Address]
 
 
-@pytest.fixture()
-def badips_map():
-  return {
-    '1.1.1.1': 'Source A',
-    '4.5.6.7': 'Best IOCs',
-    '100.12.34.56': 'Threatbusters',
-  }
+@dataclass
+class IpPacket:
+  timestamp: str
+  ip_address: IpType
 
 
-def test_good_ip_is_ok(badips_map):
-  detected = network_capture.detect_bad_ips('0\t127.0.0.1\t1.2.3.4', badips_map)
-  assert not detected
-
-
-def test_bad_ip_is_logged(badips_map):
-  detected = network_capture.detect_bad_ips('1\t4.5.6.7\t127.0.0.1', badips_map)
-  assert detected
+@dataclass
+class TlsConnection:
+  timestamp: str
```

Comment:
Make this a datetime

Reply:
This one comes from the string we get from tshark output.  We log it and don't do any calculation/duration with it, it would just get converted back into a string.  Rather than encode just to later decode, I'd rather keep it a string until we have a reason to use the date/time semantics of it.  We're not likely to do any in-process calculation with this, and we can convert it into a DATE type if storing it in a DB later, or if we use it in calculating duration later.